usaspending_api/common/recipient_lookups.py

```diff
@@ -0,0 +1,59 @@
+from usaspending_api.recipient.models import RecipientLookup
+
+
+def obtain_recipient_uri(recipient_name, recipient_unique_id, parent_recipient_unique_id=None):
+    if not recipient_unique_id:
+        return create_recipient_uri_without_duns(recipient_name, recipient_unique_id, parent_recipient_unique_id)
+
+    return fetch_recipient_hash_using_name_and_duns(recipient_name, recipient_unique_id, parent_recipient_unique_id)
+
+
+def create_recipient_uri_without_duns(recipient_name, recipient_unique_id, parent_recipient_unique_id=None):
+    recipient_hash = generate_missing_recipient_hash(recipient_name, recipient_unique_id)
+    recipient_level = obtain_recipient_level({"duns": recipient_unique_id, "parent_duns": parent_recipient_unique_id})
+    return combine_recipient_hash_and_level(recipient_hash, recipient_level)
+
+
+def fetch_recipient_hash_using_name_and_duns(recipient_name, recipient_unique_id, parent_recipient_unique_id):
+    recipient_hash = fetch_recipient_hash_using_duns(recipient_unique_id)
+    recipient_level = obtain_recipient_level({"duns": recipient_unique_id, "parent_duns": parent_recipient_unique_id})
+    return combine_recipient_hash_and_level(recipient_hash, recipient_level)
+
+
+def generate_missing_recipient_hash(recipient_unique_id, recipient_name):
+    # SQL: MD5(UPPER(CONCAT(awardee_or_recipient_uniqu, legal_business_name)))::uuid
+    import hashlib
+    import uuid
+
+    h = hashlib.md5("{}{}".format(recipient_unique_id, recipient_name).upper().encode("utf-8")).hexdigest()
+    return str(uuid.UUID(h))
+
+
+def fetch_recipient_hash_using_duns(recipient_unique_id):
+    recipient = (
+        RecipientLookup.objects.filter(duns=recipient_unique_id)
+        .values("recipient_hash")
+        .first()
+    )
+
+    return recipient["recipient_hash"] if recipient else None
+
+
+def obtain_recipient_level(recipient_record: dict) -> str:
+    if recipient_is_standalone(recipient_record):
+        level = "R"
+    elif recipient_is_child(recipient_record):
+        level = "C"
+    return level
```

Comment:
So `recipient_is_child` is the boolean opposite of `recipient_is_standalone`.  Why not just have an `else`?  It also makes my linter scream because `level` is not initialized.

Reply:
> FUNCTIONS SHOULD DO ONE THING. THEY SHOULD DO IT WELL. THEY SHOULD DO IT ONLY.

https://www.goodreads.com/book/show/3735293-clean-code, chapter 3

I'm slowly learning. Open to discussions

Reply:
Not sure I understand the reference, but the code should work.  It just seems weird to say:
```
if my_thing is one of two values:
    do this
else if my_thing is the other of two values:
    do something else
```

rather than

```
if my_thing is one of two values:
    do this
else:
    do something else
```

This also absolves you of having to initialize `level`.

¯\\\_(ツ)\_/¯